src/Behat/MinkExtension/ServiceContainer/Driver/AppiumFactory.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the Behat MinkExtension.
+ * (c) Shashikant jagtap <shashikant.jagtap@aol.co.uk>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\MinkExtension\ServiceContainer\Driver;
+
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+
+class AppiumFactory extends Selenium2Factory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDriverName()
+    {
+        return 'appium';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(ArrayNodeDefinition $builder)
+    {
+        $builder
+            ->children()
+                ->scalarNode('browser')->defaultValue('remote')->end()
+                ->append($this->getCapabilitiesNode())
+            ->end()
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildDriver(array $config)
+    {
+        $host = '0.0.0.0:4723';
+
+        $config['wd_host'] = sprintf('%s/wd/hub', $host);
+
+        return parent::buildDriver($config);
+    }
+
+    protected function getCapabilitiesNode()
+    {
+        $node = parent::getCapabilitiesNode();
+
+        $node
+            ->children()
+            ->scalarNode('automationName')->defaultValue('Appium')->end()
```

Comment:
please indent the calls inside the ``->children()`` node to better reflect the levels in the tree. It makes things much more readable